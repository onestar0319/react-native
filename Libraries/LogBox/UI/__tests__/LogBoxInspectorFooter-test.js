/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const LogBoxInspectorFooter = require('../LogBoxInspectorFooter').default;
const render = require('../../../../jest/renderer');

describe('LogBoxInspectorFooter', () => {
  it('should render two buttons for warning', () => {
    const output = render.shallowRender(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="warn"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render two buttons for error', () => {
    const output = render.shallowRender(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="error"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render two buttons for fatal', () => {
    const output = render.shallowRender(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="fatal"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render no buttons and a message for syntax error', () => {
    const output = render.shallowRender(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="syntax"
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
