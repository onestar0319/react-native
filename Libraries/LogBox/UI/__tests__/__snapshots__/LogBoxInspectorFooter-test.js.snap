// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorFooter should render fatal button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(51, 51, 51, 1)",
        "elevation": 1,
        "flexDirection": "row",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
      },
      Object {
        "shadowColor": "rgba(34, 34, 34, 1)",
      },
    ]
  }
>
  <LogBoxButton
    backgroundColor={
      Object {
        "default": "rgba(243, 83, 105, 1)",
        "pressed": "rgba(208, 75, 95, 1)",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 14,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Reload
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 11,
            "includeFontPadding": false,
            "lineHeight": 12,
          }
        }
      >
        Fatal
         errors require a full reload
      </Text>
    </View>
    <ForwardRef(SafeAreaView) />
  </LogBoxButton>
</View>
`;

exports[`LogBoxInspectorFooter should render fatal button for syntax erorr 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(51, 51, 51, 1)",
        "elevation": 1,
        "flexDirection": "row",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
      },
      Object {
        "shadowColor": "rgba(34, 34, 34, 1)",
      },
    ]
  }
>
  <LogBoxButton
    backgroundColor={
      Object {
        "default": "rgba(243, 83, 105, 1)",
        "pressed": "rgba(208, 75, 95, 1)",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 14,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Reload
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 11,
            "includeFontPadding": false,
            "lineHeight": 12,
          }
        }
      >
        Syntax
         errors require a full reload
      </Text>
    </View>
    <ForwardRef(SafeAreaView) />
  </LogBoxButton>
</View>
`;

exports[`LogBoxInspectorFooter should render fatal for warning with a fatal 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(51, 51, 51, 1)",
        "elevation": 1,
        "flexDirection": "row",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": -2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
      },
      Object {
        "shadowColor": "rgba(34, 34, 34, 1)",
      },
    ]
  }
>
  <LogBoxButton
    backgroundColor={
      Object {
        "default": "rgba(243, 83, 105, 1)",
        "pressed": "rgba(208, 75, 95, 1)",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 14,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
        Reload
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.8)",
            "fontSize": 11,
            "includeFontPadding": false,
            "lineHeight": 12,
          }
        }
      >
        Syntax
         errors require a full reload
      </Text>
    </View>
    <ForwardRef(SafeAreaView) />
  </LogBoxButton>
</View>
`;

exports[`LogBoxInspectorFooter should render two buttons for warning with no fatal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "elevation": 1,
      "flexDirection": "row",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": -2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <FooterButton
    onPress={[Function]}
    text="Minimize"
  />
  <FooterButton
    onPress={[Function]}
    text="Dismiss"
  />
</View>
`;
